modules/types/src/array-types.ts

```diff
@@ -40,4 +40,154 @@ export type NumericArray = TypedArray | number[];
  * TypeScript type covering all typed arrays and classic arrays consisting of numbers
  * @note alias for NumericArray
  */
-export type NumberArray = NumericArray;
+export type NumberArray = TypedArray | number[];
+
+/** Array with exactly 1 number */
+export type NumberArray1 = [number];
+
+/** Array with exactly 2 numbers */
+export type NumberArray2 = [number, number];
+
+/** Array with exactly 3 numbers */
+export type NumberArray3 = [number, number, number];
+
+/** Array with exactly 4 numbers */
+export type NumberArray4 = [number, number, number, number];
+
+/** Array with exactly 6 numbers */
+export type NumberArray6 = [number, number, number, number, number, number];
+
+/** Array with exactly 8 numbers */
+export type NumberArray8 = [number, number, number, number, number, number, number, number];
+
+/** Array with exactly 9 numbers */
+export type NumberArray9 = [number, number, number, number, number, number, number, number, number];
+
+/** Array with exactly 12 numbers */
+export type NumberArray12 = [
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number
+];
+
+/** Array with exactly 16 numbers */
+export type NumberArray16 = [
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number
+];
+
+/**
```

Comment:
Why are all these declarations being added?

Reply:
> Why are all these declarations being added?

These came from gl-matrix. I am still evaluating various approaches to typing these things. But removed them for now.